### ISSUE TITLE
Deploying on shinyapps.io: A package dependency should not be required, yet Shiny asks for it

### ISSUE THREAD

**Issue by adeldaoud** [rstudio/shiny#2527]
Hi

I am building a Shiny application based on an unpublished package. I have, therefore, explicitly created stand-alone functions of parts of this unpublished package. Shiny call these functions when needed. On my local machine, Shiny is not calling this unpublished package. I have even uninstalled this package to make sure that Shiny is independently functioning. 

Yet, when I deploy this package to shinyapps.io, I face an error about this unpublished package is required.

 I suspected that an earlier Shiny deployment of this app might have interfered and that this dependencies is an echo. So I removed that earlier version to make sure that this app is not interfering with the new app. But the same problem persists.

In summary, a package dependency should not be required, yet Shinyapps asks for it.Any idea of what is going on? 

Not sure how to write a reproducible example of this particular issue. 

(NB: I have posted this question here as well: https://community.rstudio.com/t/deploying-on-shinyapps-io-a-package-dependency-should-not-be-required-yet-shiny-asks-for-it/35517?u=adeldaoud)

**Comment by cpsievert** [comment]
Does your app contain anything like `library(pkgname)`; `require(pkgname)`; or `pkgname::` of the package in question? The mechanism that deploys the app will look for things like this to automatically infer what packages are needed.

**Comment by adeldaoud** [comment]
@cpsievert , thanks for the input. No, I am not using pkgname but your input made me think about something else. I do have a "TEMP" folder in the "myapp" folder where I store old functions of which one referred to that unpublished package. Matter solved!
